### PR TITLE
Projects

### PR DESCRIPTION
**Pull request by gvdent**
Introduces the project registration to weaver-server

### CODE REVIEW

File: config/custom-environment-variables.coffee [hunk 1]
```diff
@@ -1,4 +1,9 @@
 module.exports =
   server:
     weaver:
-      port: "WEAVER.PORT"
\ No newline at end of file
+      port: "WEAVER_PORT"
+
+  services:
+    project:
+      endpoint: "SERVICES_PROJECT_ENDPOINT"
```
Comment (malamili) [inline config/custom-environment-variables.coffee]: here endpoint is small case, in default its camelcase

File: src/comm/HttpComm.coffee [hunk 2]
```diff
@@ -7,18 +7,32 @@ class HTTP
...
+     @routeHandler.getRoutes.forEach((route) =>
+       app.get(@transform(route), (req, res) =>
+
+         req.payload = req.query['payload']
+         req.payload = "{}" if not req.payload?
+
+         try
+           req.payload = JSON.parse(req.payload)
+         catch error
+           res.status(400).send("Invalid json: #{error}")
```
Comment (malamili) [inline src/comm/HttpComm.coffee]: There is a better way of sending errors. See for instance CreateNode.coffee in operations folder

File: src/project/ProjectCtrl.coffee [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+bus.on('project.create', (req, res) ->
+  if !req.payload.name?
+    res.error("Expects name parameter")
+  else
+    doCall(res, "create/#{req.payload.name}")
+)
+
+bus.on('project.delete', (req, res) ->
+  if !req.payload.id?
+    res.error("Expects id parameter")
```
Comment (malamili) [inline src/project/ProjectCtrl.coffee]: Same as above (you can also make a new error code in commons)

File: src/project/ProjectCtrl.coffee [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  res.promise(
+    rp({
+      uri: createUri(suffix)
+    })
+  )
+
+bus.on('project', (req, res) ->
+  doCall(res, "list")
+)
+
```
Comment (malamili) [inline src/project/ProjectCtrl.coffee]: May I propose a nicer way of chaining this using function overloading?

```
doCall = (suffix) -> (req, res) -> 
   res.promise(
     rp({
       uri: createUri(suffix)
     })
   )
 
 bus.on('project', doCall("list"))
```

File: src/routes.coffee [hunk 5]
```diff
@@ -8,5 +8,10 @@ WEAVER.GET   "application.version"    # Application version
 WEAVER.GET   "read"                   # Reads a single entity
 WEAVER.POST  "write"                  # Execute Create, Update and Delete operations in bulk
 
+WEAVER.GET  "project"                 # Get a list of projects
```
Comment (malamili) [inline src/routes.coffee]: How about a project.update for for instance changing its name? or do you always get ID's

File: src/comm/HttpComm.coffee [hunk 6]
```diff
@@ -16,6 +16,11 @@ class HTTP
          req.payload = req.query['payload']
          req.payload = "{}" if not req.payload?
 
+         try
```
Comment (malamili) [inline src/comm/HttpComm.coffee]: viespeuk!

File: src/project/ProjectCtrl.coffee (hunk 4 above)
Comment (gvdent) [inline src/project/ProjectCtrl.coffee]: Nice, changed

File: src/project/ProjectCtrl.coffee (hunk 3 above)
Comment (gvdent) [inline src/project/ProjectCtrl.coffee]: Code changed entirely. One issue with the commons error codes: They are websocket specific and don't map well to rest :(

File: src/routes.coffee (hunk 5 above)
Comment (gvdent) [inline src/routes.coffee]: Getting rid of name in a next iteration of the service.

File: src/comm/HttpComm.coffee (hunk 2 above)
Comment (malamili) [inline src/comm/HttpComm.coffee]: `res.status(400).send("{"code":"32", "message:":"Invalid json: #{error}"}")`

Comment (gvdent) [inline src/comm/HttpComm.coffee]: Fixed

File: config/custom-environment-variables.coffee (hunk 1 above)
Comment (gvdent) [inline config/custom-environment-variables.coffee]: removed camelcase in config/default.coffee

**Pull request merged**